### ISSUE TITLE
list of errors returned by nedb?

### ISSUE THREAD

**Issue by yegodz** [louischatriot/nedb#243]
Hi..
I am trying to use `verror` to propagate errors returned by `nedb`. For instance, 

    my_insert(fileDoc, callback) { 
     nedb.insert(fileDoc, function(err, newDoc) {
            if (err) {
              // console.log('nedb returned' + require('util').inspect(err)); 
              return callback(new VError(err, 'dbinsert failed'));
            }
     .....
      }

`VError` allows me to wrap a new error around the error returned from `nedb` before propagating it up the chain. 

This should work if the `err` returned by `nedb.insert` is of type `Error` or `VError`, but it seems to be a private `Object`.
I now need to parse the list of possible error codes returned by nedb.. but I cannot find them in any of the `.js` files.

Can you point me to where you have declared the format of your returned errors as well as the full list of return codes like `uniqueViolated`?

**Comment by anied** [comment]
I tend to agree that there is perhaps an issue here.  I believe Node convention is that an error returned to a callback should be an instance of `Error` -- however, I also am seeing that errors returned to the callback in nedb's `insert` method are instead simply javascript `Object` instances, _not_ instances of `Error`.

I would submit the following links to support my assertion that I would expect to see an `Error`:
https://github.com/google/google-api-nodejs-client/issues/345
https://github.com/addyosmani/psi/issues/43

My apologies if this is incorrect, and for piggybacking on an existing bug, but I do think that this warrants review.  Let me know if I can be of assistance.
